common/cmake/devices.cmake

```diff
@@ -1472,8 +1466,9 @@ function(ADD_FPGA_TARGET)
     )
 
   if(NOT "${ADD_FPGA_TARGET_SDC_FILE}" STREQUAL "")
```

Comment:
Shouldn't we make this part dependent on whether we have the INPUT_XDC_FILE and add `--sdc_file ${OUT_SDC}`
I mean where do we add the dependency of VPR on the generated SDC?

Reply:
Sorry, can you rephrase your statement.  I'm not sure I understand what you are saying.

Reply:
It looks to me that we are not passing the generated SDC to VPR, but the one specified as SDC_FILE in the CMakeLists.txt. We should have something akin:
```
    append_file_dependency(VPR_DEPS ${OUT_SDC_REL})
    set(SDC_ARG --sdc_file ${OUT_SDC})
```

Reply:
Fixed the dependencies. Now, there are three possibilities:

- NO SDC, and only the XDC: dependencies added on the generated SDC file
- NO XDC, and only the SDC: dependencies added on the input SDC file
- SDC and XDC: dependencies added on both the generated and input SDC, but the SDC that is going to be used is the user-provided one instead of the autogenerated one.

Reply:
> NO XDC, and only the SDC: dependencies added on the input SDC file

Sorry, do you mean that the input SDC is modified?  Or do you mean that a new SDC is generated with the propigated outputs?

Reply:
This means that the SDC is just read, without propagation and it needs to be read manually.